### DISCUSSION TITLE
Zwangsladung über die Entladesteuerung

### DISCUSSION THREAD

**Discussion by iReckyy** [wiggal/GEN24_Ladesteuerung#111] [Q&A] [answered]
Ich habe heute die Zwangsladung ausprobiert, um den Akku gleichmäßig mit 2KW über den Tag voll zu laden.
Leider ist bei mir die PV-Leistung jetzt schon zu niedrig für den Eigenverbrauch und die 2KW für den Akku. Er hört dann einfach auf mit der Zwangsladung und entlädt den Akku wieder. 

Das ist schon etwas komisch. 

Im Gen24 stehen die Max Ladegrenze und die Min Ladegrenze drin, trotzdem macht er etwas anderes?

Aus dem Netz Laden habe ich erlaubt.

Würde er bei der Zwangsladung die Akkuschonung berücksichtigen?

**Comment by wiggal** [comment]
habe es gerade bei mir ausprobiert.
2000 Watt Mindesladung im Gen24 eingetragen. 
Er lädt tatsächlich mit 2000Watt 500 Watt PV Rest aus Netz.

Es ist ja für den Gen24 nur massgebend was Im Batteriemanagement steht, also wenn Minladung 2000 dann muss er die auch laden.

Welche Firmware hast du am Gen24?

**Reply by wiggal** [reply]
ja gut meine Maxladung war in dem Beispiel 5000W. Vielleicht schlägt hier Maxladung die Minladung. Müsste man dann abfangen

**Reply by wiggal** [reply]
Weißt du, was im Batteriemanagement stand, als er mit der Zwangsladung aufhörte, bzw. hie hoch der Akkustand war?
Hast du den zugehörigen Auszug aus dem Logfile?

**Reply by wiggal** [reply]
OK, daran hapert es:

Wenn Min. Ladeleistung größer als Max. Ladeleistung ist, trägt er die Werte zwar ein (was komisch ist, über die Oberfläche gehts nicht). Es taucht dann auch der Hinweis auf den Fehler auf, die Ladebegrenzungen funktionieren aber nicht mehr.

![grafik](https://github.com/user-attachments/assets/77c3b4a6-9e55-4f44-8b7a-5c9fabea0c08)

Habe es gefixed, kommt mit der morgigen Version.

Danke für die Info

**Reply by wiggal** [reply]
Ist raus, wenn du es mal getestet hast, kannst du es ja als beantwortet markieren.
Danke

**Reply by iReckyy** [reply]
Werde ich bei Zeiten erneut testen.

Danke für den schnellen Fix.

**Reply by wiggal** [reply]
Hast du die Zwangsladung schon wieder mal getestet?

**Reply by iReckyy** [reply]
Noch nicht, habe es aber eben mal angeworfen.

Werde berichten.

**Reply by iReckyy** [reply]
Funktioniert.

Ist nur doof, wenn der Akku noch nicht voll ist, wenn die Sonne weg ist.

Dann greift die Akkuschonung nicht mehr.

**Reply by wiggal** [reply]
Bei der Zwangsladung dürfte die Akkuschonung, sowieso nicht greifen

**Reply by iReckyy** [reply]
Hmm, das wäre aber schon schön 😃

**Reply by wiggal** [reply]
Ist halt an einer ganz ander Stelle, Im Entlademanagement nicht im Lademanagement. Mal sehen.

**Reply by wiggal** [reply]
Ich habe jetzt mal eingebaut, dass auch bei einer Zwangsladung eine Akkuschonung konfiguriert werden kann.
Dazu musst du Batterieentlandung_steuern = 2 setzen.
Wiggal

**Comment by iReckyy** [comment]
Wie es scheint, funktioniert die Zwangsladung aktuell garnicht.
Er schreibt mir zwar die Max Ladung, aber nicht die Min Ladung.

**Reply by wiggal** [reply]
Ja, wenn man mal schnell was am Code macht. 
Ich hoffe, jetzt klappt’s.

**Comment by iReckyy** [comment]
Hallo,

ich habe heute mal wieder die Zwangsladung zwecks Akkupflege benutzt.
>
Irgendwie trägt er mir da sehr komsiche Werte ein.
>
Anfangs hat es mehr oder weniger noch gepasst. Ladewert x SOH, also etwas weniger.
Bei der Zwangsladung ist der SOH mMn nicht unbedingt erforderlich.
>
Aber später hat er ganz wirres Zeug geschrieben. Siehe Log.
Die Akkuschonung habe ich zu Testzwecken in meiner Winter_Priv.ini aus.
>
[Crontab.log](https://github.com/user-attachments/files/24647365/Crontab.log)
>
>
Edit. Habe selber mal den Log durchgesehen, ich glaube beim Zwangsladen ist die Akkuschonung unabhängig des Schalters in der *_Priv.ini eingeschaltet.

**Reply by wiggal** [reply]
Mit den Infos kann ich leider nichts anfangen.
> ich habe heute mal wieder die Zwangsladung zwecks Akkupflege benutzt.
Irgendwie trägt er mir da sehr komsiche Werte ein.

Welche Werte wo?

> Anfangs hat es mehr oder weniger noch gepasst. Ladewert x SOH, also etwas weniger.
Bei der Zwangsladung ist der SOH mMn nicht unbedingt erforderlich.

Ladewert x SOH, was meist du damit?

> Aber später hat er ganz wirres Zeug geschrieben. Siehe Log.
Die Akkuschonung habe ich zu Testzwecken in meiner Winter_Priv.ini aus.

Welches wirres Zeug, die Crontab.log hat über 95 tausend Zeilen, wie soll ich da finden, was du meinst.
Kannst du die Informationen bitte genauer formulieren?

Danke Wiggal

**Reply by iReckyy** [reply]
Ich habe in der ENTLadeStrg die Zwangsladewerte eingetragen. Mal -3KW, dann -2KW.
>
![Zwangsladung](https://github.com/user-attachments/assets/be82daa6-4853-486f-9cc4-923868a3047c)
>
Sieht man in jedem Einzellog der letzten 2-3h.
>
***** BEGINN:  2026-01-15 16:46:01 *****

API von Symo-WR  192.168.178.21  nicht verfügbar, Ersatz-AC-Wert loggen!
Fehlerursache: 'PowerReal_PAC_Sum'

API_URL zur Fehlersuche:  http://192.168.178.21/components/readable
Offline_AC_DIFF:  -3
Offline_AC_DIFF nicht kleiner 0:  0 

DEBUG Parameteränderung durch WebUI-Settings: Ladesteuerung:['logging', 'laden', 'entladen', 'optimierung', 'notstrom']
Programmversion:             v0.41.0
aktuellePrognose:            114
TagesPrognose - BattVollUm:  0 - 13
Grundlast_Summe für Tag:     0
aktuellePVProduktion/Watt:   1
aktuelleEinspeisung/Watt:    -751
aktuelleBatteriePower/Watt:  -2
GesamtverbrauchHaus/Watt:    750
aktuelleBattKapazität/Watt:  0
Batteriestatus in Prozent:   100.0 %
LadewertGrund:  Nicht schreiben, da PVProduktion < 50 Watt!
Bisheriger Ladewert/Watt:    5000
Neuer Ladewert/Watt( 0.9):   5000

## ENTLADESTEUERUNG ##

Feste Entladegrenze in % :  30 %
VerbrauchsgrenzeEntladung:  0.0 W
Feste Entladegrenze Table:  -2000 W
Zwangsladung ALT:           1024 W
Zwangsladung NEU:           1024 W
Ladewert >= Zwangsladung:    5000 W

Batteriesteuerung: Änderungen kleiner Schreibgrenzen!

## Eigenverbrauchs-Optimierung ##
Prognose nächste 24Std:  7.626 KW
Eigenverbrauchs-Opt. ALT:  0 W
Eigenverbrauchs-Opt. NEU:  0 W

Eigenverbrauchs-Opt.: Keine Änderung!

In SQLite-Datei gespeichert!

DEBUG *************** Berechnung Abzugswert: 
DEBUG
DEBUG *************** Prognosemittel LOOP: 0
DEBUG ########### Pro_Akt: 76 REST_Akt_Minute: 0.23
DEBUG *************** Prognosemittel LOOP: 1
DEBUG ########### Pro_Akt: 38 Akt_Minute: 0.77
DEBUG  2026-01-15 17:00:00 Pro_Akt: 38 Prognose_gesamt: 114
DEBUG
DEBUG >> 
DEBUG Akkuschonung_Werte: {'80': '0.25', '85': '0.2', '90': '0.15', '99': '0.1'}
DEBUG AkkuschonungLadewert-alterLadewert: 3976
DEBUG BattStatusProz_Grenze: 99
DEBUG AkkuschonungLadewert: 1024
DEBUG aktuellerLadewert: 5000
**DEBUG Keine Akkuschonung, da in LadeStrg abgewählt!**

DEBUG Keine Begrenzung, da Akkuschonung in LadeStrg abgewählt!DEBUG
DEBUG BattVollUm:                 13Uhr
DEBUG WRSchreibGrenze_nachUnten:  607W
DEBUG WRSchreibGrenze_nachOben:   607WDEBUG
DEBUG >>>>>>
DEBUG MaxEntladung = entladesteurungsdata:2703
DEBUG VerbrauchsgrenzeEntladung aus Spalte 1: 0.0
DEBUG *************** Berechnung Abzugswert: 
DEBUG
DEBUG *************** Prognosemittel LOOP: 0
DEBUG ########### Pro_Akt: 76 REST_Akt_Minute: 0.23
DEBUG *************** Prognosemittel LOOP: 1
DEBUG ########### Pro_Akt: 38 Akt_Minute: 0.77
DEBUG  2026-01-15 17:00:00 Pro_Akt: 38 Prognose_gesamt: 114

DEBUG FesteEntladegrenze index 0 = -2000
DEBUG FesteEntladegrenze aus Spalte 2: -2000
**DEBUG Akkuschonung auch bei Zwangsladung: 1024**
DEBUG Batterieentladegrenze NEU: 1024%
DEBUG >>>>>>
DEBUG
DEBUG >>>>>>
DEBUG ## Dauer_Nacht_Std: 16.23, Akku_Rest_Watt: 761
DEBUG ## Eigen_Opt_genau: 46, Eigen_Opt_Std_neu: 0
DEBUG
DEBUG >>

***** ENDE:  2026-01-15 16:46:02 *****
>
Habe es mal Fett markiert.

**Reply by wiggal** [reply, marked as answer]
Was hast du denn in der CONFIG/charge_priv.ini unter [Entladung] stehen:

```
; Konfigdaten für die Entladesteuerung
; Batterieentladung und Zwangsladung ein- bzw. ausschalten (1=ein, 0=aus, 2=Ein+Akkuschonung bei Zwangsladung)
Batterieentlandung_steuern = 0
```

Vermutlich hast du Batterieentlandung_steuern = 2, oder?

**Reply by iReckyy** [reply]
S

> Was hast du denn in der CONFIG/charge_priv.ini unter [Entladung] stehen:
> 
> ```
> ; Konfigdaten für die Entladesteuerung
> ; Batterieentladung und Zwangsladung ein- bzw. ausschalten (1=ein, 0=aus, 2=Ein+Akkuschonung bei Zwangsladung)
> Batterieentlandung_steuern = 0
> ```
> 
> Vermutlich hast du Batterieentlandung_steuern = 2, oder?

Schande über mein Haupt, ja…
>
Aber der SOH wird für den Ladewert schon genutzt, oder?
>
Weil 3KW x 0.92 = 2,76KW.

**Reply by wiggal** [reply]
Was meinst du mit der Formel, kannst du den Zusammenhang zu welchen Ladewert erläutern?

**Reply by iReckyy** [reply]
Wenn ich 3KW bei der Zwangsladung eingebe, werden 2,76KW im Gen24 bei der Zeitsteuerung eingegeben.
>
Das entspricht exakt dem Sollwert von 3KW multipliziert mit dem SOH des Akkus von 0,92, bzw 92%.

**Reply by wiggal** [reply]
Nein, wenn mann 3kW eingibt, also 3kW will, bekommt man die auch, alles andere wäre nicht nachvollziehbar.
Das einzige, was ich schon mal überlegt habe, aber nicht umgesetzt habe, wäre die Akkukapazität für die Berechnungen auf den SOH anzupassen.

**Reply by wiggal** [reply]
OK, ich hab nochmal nachgesehen, der SOH wird bei der Akkukapazität berücksichtigt.
Ich hab es jetzt mal in der Hilfe zum `config-TAB´ :
```
; Akkuschonung_Werte = {"Ladestand%" : "LadewertC", ...}
; Akkuschonung_Ladewert = ((Akkukapazität * SOH / 100) * LadewertC), aufgerundet auf ganze 10 Watt.
; Heißt, der Akkuzustand wird hier wie auch bei anderen Berechnungen angehalten. 
```

**Reply by iReckyy** [reply]
> Was hast du denn in der CONFIG/charge_priv.ini unter [Entladung] stehen:
> 
> ```
> ; Konfigdaten für die Entladesteuerung
> ; Batterieentladung und Zwangsladung ein- bzw. ausschalten (1=ein, 0=aus, 2=Ein+Akkuschonung bei Zwangsladung)
> Batterieentlandung_steuern = 0
> ```
> 
> Vermutlich hast du Batterieentlandung_steuern = 2, oder?

Habe jetzt mal mit Batterieentlandung_steuern = 1 getestet um ohne Akkuschonug zu laden, da gibt er mir einen Fehler aus:
>

Traceback (most recent call last):
File "/home/GEN24/EnergyController.py", line 395, in
if (AkkuschonungLadewert < Neu_BatteryMaxDischarge and Batterieentlandung_steuern > 1): Neu_BatteryMaxDischarge = AkkuschonungLadewert
^^^^^^^^^^^^^^^^^^^^
NameError: name 'AkkuschonungLadewert' is not defined. Did you mean: 'Akkuschonung_Werte'?
>
[Crontab.log](https://github.com/user-attachments/files/24719389/Crontab.log)

**Reply by wiggal** [reply]
war ein BUG:
ich hoffe ich habe ihn mit a3a8fc3 gefixt
